### PR TITLE
[#61] Add GitHub add-on

### PR DESCRIPTION
**Pull request by hoangmirs**
- Resolves https://github.com/nimblehq/infrastructure-templates/issues/61
- Resolves https://github.com/nimblehq/infrastructure-templates/issues/60

## What happened 👀

- Create GitHub addon:
  - Add GitHub templates
  - Add `lint` workflow
  - Create `.gitignore` from [this template](https://github.com/github/gitignore/blob/main/Terraform.gitignore)

## Insight 📝

- As dotfiles are not shown by default from `glob.sync`, I need to add the option `dot: true`
- Add `toBeEmpty` matcher to test if the folder is empty

## Proof Of Work 📹

Tests should be passed

```bash
nimble-test
├── .github
│   ├── ISSUE_TEMPLATE
│   ├── PULL_REQUEST_TEMPLATE
│   ├── PULL_REQUEST_TEMPLATE.md
│   └── workflows
├── .gitignore
├── main.tf
├── modules
│   ├── alb
│   ├── bastion
│   ├── ecr
│   ├── ecs
│   ├── log
│   ├── rds
│   ├── s3
│   ├── security_group
│   ├── ssm
│   └── vpc
├── outputs.tf
├── providers.tf
└── variables.tf
```

### CODE REVIEW

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+https://github.com/nimblehq/git-template/issues/??
```
Comment (malparty) [inline skeleton/github/.github/PULL_REQUEST_TEMPLATE.md:1]: Should we use the `Close` keyword here? :D 

```suggestion
- Close #
```

If you agree, please approve this PR: https://github.com/nimblehq/git-template/pull/32 😇

File: src/commands/generate/index.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 import { detectTerraform, formatCode } from '../../helpers/terraform';
+import {
+  applyVersionControl,
+  versionControlChoices,
+} from '../../templates/addons/versionControl';
 import { generateAwsTemplate } from '../../templates/aws';
 
 type GeneralOptions = {
   projectName: string;
+  versionControl?: 'github' | 'gitlab';
```
Comment (malparty) [inline src/commands/generate/index.ts:14]:
```suggestion
  versionControl?: 'github' | 'none';
```
(or eventually "other")

image

File: src/templates/addons/versionControl/index.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    choices: [
+      {
+        value: 'github',
+        name: 'GitHub',
+      },
+      {
+        value: 'gitlab',
+        name: 'GitLab (NOT IMPLEMENTED YET)',
+        disabled: true,
+      },
```
Comment (malparty) [inline src/templates/addons/versionControl/index.ts:18]: Related to my MEME comment.
IMO we should remove ALL the "not implemented yet" options because they are noise. 
Instead, we can eventually add the issues to GitHub.
If something is not implemented, we should not suggest it as an option.

It feels like 
- "Good morning! Do you want a coffee?"
- "Oh yeah, thanks!"
- "Because the machine is broken!" :harold:

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (hoangmirs) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE.md:1]: I agree. But you should mark that PR ready for review 😄

File: src/templates/addons/versionControl/index.ts (hunk 3 above)
Comment (hoangmirs) [reply src/templates/addons/versionControl/index.ts:18]: Makes sense. I will remove them and create stories for them instead 🙏

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (hoangmirs) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE.md]: Added in https://github.com/nimblehq/infrastructure-templates/pull/97/commits/2963dbf060c1f027293ed8f05ef9b51d45d512a9

File: src/commands/generate/index.ts (hunk 2 above)
Comment (hoangmirs) [reply src/commands/generate/index.ts]: 😅 Removed in https://github.com/nimblehq/infrastructure-templates/pull/97/commits/61b304a1139e79660f96902fa0b5f539be5eab9f

File: src/templates/addons/versionControl/index.ts (hunk 3 above)
Comment (hoangmirs) [reply src/templates/addons/versionControl/index.ts]: Removed them in https://github.com/nimblehq/infrastructure-templates/pull/97/commits/becaba37870fac88ad0bb1645d76b9cf8d41b239

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-https://github.com/nimblehq/git-template/issues/??
+Close #
```
Comment (malparty) [inline skeleton/github/.github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
- Close #
```

|Without the dash| With the dash |
|---|---|
|image|image|

**Review by malparty — APPROVED** [review]
A last minor suggestion ⬆️ 
But the rest is awesome!! Thanks for having also cleaned the previous options (Heroku, etc...) 🙇 🙇 🙇 🙇

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (hoangmirs) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE.md]: 💯 That's good, changed in https://github.com/nimblehq/infrastructure-templates/pull/97/commits/c3fe87984b2aadd8b57c25720be25114d27ee4eb

**Review by malparty — APPROVED** [review]

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+- Close #
```
[the remark this answers is not in the document]
Comment (malparty) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE.md:1]: Yeah, sorry I'm just ahead of time :D This is the `new new PR template` 😇 
https://github.com/nimblehq/git-template/pull/32/files

[the remark this answers is not in the document]
Comment (malparty) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE.md:1]: Let update the content descriptions but keep the `- Close #` (as there is a PR in progress) ;-)

[the remark this answers is not in the document]
Comment (hoangmirs) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE.md:1]: Thanks, changed in https://github.com/nimblehq/infrastructure-templates/pull/97/commits/f3a5013f21782f897d447d2e6383c6c8e4ad924c

File: skeleton/github/.github/PULL_REQUEST_TEMPLATE/release_template.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+or
+Link to the project management tool for the release
+
+## Features
+
+Provide the ID and title of the issue in the section for each type (feature, chore and bug). The link is optional.
+
+- [sc-1234] As a user, I can log in
+or just
+- [[sc-1234](https://app.shortcut.com/nimblehq/story/1234)] As a user, I can log in
```
[the remark this answers is not in the document]
Comment (hoangmirs) [reply skeleton/github/.github/PULL_REQUEST_TEMPLATE/release_template.md]: Thanks, changed in https://github.com/nimblehq/infrastructure-templates/pull/97/commits/f3a5013f21782f897d447d2e6383c6c8e4ad924c

**Comment by hoangmirs** [comment]
@malparty @andyduong1920 I requested you to review this PR again as I changed a bit:
1. Add `None` option for version control in case we don't use it
2. Move `skeleton/github` into `skeleton/addons/versionControl/github`

Please help me take a look at that 🙏

**Review by malparty — APPROVED** [review]
LGTM! :)

**Review by longnd — APPROVED** [review]
